src/include/migraphx/shape.hpp

```diff
@@ -296,6 +296,11 @@ struct MIGRAPHX_EXPORT shape
     /// not transposed.
     bool standard() const;
 
+    /// Returns true if the shapes are compatible.  TODO: better description
+    // Paul, How would you describe the purpose of that equality check for the shapes?  I'm trying to come up with a 
+    // function description that explains why it's ok for the strides not to match sometimes.
+    bool compatible_lens(const shape& s2) const;
```

Comment:
The function should probably be in program.cpp file since its only used for that. Either way it should be a free function so we can make clear which is the actual shape and which is the expected shape. And should be called `is_compatible_shape`.